src/main/java/edu/umich/its/cpm/MigrationInstanceService.java

```diff
@@ -104,45 +161,113 @@ private void updateMigrationStatusAndEndTime() {
 		for (Migration migration : allOngoingMigrations)
 		{
 			String mId = migration.getMigration_id();
-			int allItemCount = fRepository.getMigrationBoxFileCountForMigration(mId);
-			int allFinishedItemCount = fRepository.getFinishedMigrationBoxFileCountForMigration(mId);
-			if (allItemCount > 0 && allItemCount == allFinishedItemCount )
+			String destination_type = migration.getDestination_type();
+			if (Utils.MIGRATION_TYPE_BOX.equals(destination_type))
 			{
-				// all the items within the migration is finished
-				// update the end time of the parent record
-				Timestamp lastItemMigrationTime = fRepository.getLastItemEndTimeForMigration(mId);
-				mRepository.setMigrationEndTime(lastItemMigrationTime, mId);
+				// for Box file migration
+				updateBoxMigrationTimeAndStatus(mId);
+			}
+			else if (Utils.MIGRATION_TYPE_GOOGLE.equals(destination_type))
```

Comment:
Can the migration_type be more specific?  E.g. MIGRATION_TYPE_GOOGLE_GROUP (or EMAIL)?  This doesn't do all things Google and some user might make a case that the resources should go to Google Drive.  If so that would require going back and changing code.